### PR TITLE
Fix attrib storage on changed value

### PR DESCRIPTION
**Pull request by adrianschroeter**
The backend got not update on changed attribute values leading to a broken history

Fixes https://github.com/openSUSE/open-build-service/issues/7673

### CODE REVIEW

File: src/api/test/functional/attributes_test.rb [hunk 1]
```diff
@@ -379,9 +379,29 @@ def test_create_attributes_project
     assert_response :success
     assert_xml_tag(tag: 'revisionlist')
     revision = Xmlhash.parse(@response.body).elements('revision').last
-    assert_equal 'tom', revision['user']
+    assert_equal 'king', revision['user']
     srcmd5 = revision['srcmd5']
 
+    # check history on changed value only
```
Comment (coolo) [inline src/api/test/functional/attributes_test.rb]: sorry, but the test is called test_create_attributes_project - testing updates from within is the reason why developers loath the minitests :(

File: src/api/app/models/attrib.rb [hunk 2]
```diff
@@ -34,7 +34,7 @@ class Attrib < ApplicationRecord
            :validate_issues,
            :validate_allowed_values_for_attrib_type
 
-  after_commit :write_container_attributes, on: [:create, :destroy]
+  after_commit :write_container_attributes, on: [:create, :destroy, :update]
```
Comment (coolo) [inline src/api/app/models/attrib.rb]: this is still called explicitly in webui - and I had a reason back then, but I don't remember

Comment (vpereira) [reply src/api/app/models/attrib.rb]: probably because it wasn't being called here?

File: src/api/test/functional/attributes_test.rb (hunk 1 above)
Comment (vpereira) [reply src/api/test/functional/attributes_test.rb]: well, even if I agree with you, the same test destroy attributes to.. so I guess it's a general issue in the minitest test suite.. handling it as  nitpick.

**Pull request merged**